lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,6 +56,68 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHead = (doc) ->
+  loadedAssets = extractTrackAssets document
+  fetchedAssets = extractTrackAssets doc
+
+  updateLinkTags(loadedAssets, fetchedAssets)
+  updateScriptTags(loadedAssets, fetchedAssets)
+
+extractTrackAssets = (doc) ->
+  for node in doc.querySelector('head').childNodes when node.getAttribute?('data-turbolinks-track')?
+    node
+
+filterForNodeType = (nodeList, nodeType) ->
+  (node for node in nodeList when node.nodeName == nodeType)
+
+insertScript = (scriptNode) ->
+  newNode = document.createElement 'script'
+  newNode.setAttribute attr.name, attr.value for attr in scriptNode.attributes
+  newNode.appendChild document.createTextNode scriptNode.innerHTML
+
+  deferred = $.Deferred()
+
+  onScriptLoad = ->
+    newNode.removeEventListener('load', onScriptLoad)
+    newNode.removeEventListener('error', onScriptError)
+    deferred.resolve()
+
+  onScriptError = ->
+    newNode.removeEventListener('load', onScriptLoad)
+    newNode.removeEventListener('error', onScriptError)
+    deferred.reject()
+
+  newNode.addEventListener('load', onScriptLoad)
+  newNode.addEventListener('error', onScriptError)
+
+  document.head.appendChild(newNode)
+
+  return deferred.promise()
+
+nodeInList = (node, list) ->
+  list.some((listNode) -> node.isEqualNode(listNode))
+
+updateScriptTags = (loadedAssets, fetchedAssets) ->
+  oldScripts = filterForNodeType(loadedAssets, 'SCRIPT')
+  newScripts = filterForNodeType(fetchedAssets, 'SCRIPT')
+
+  scriptPromises = []
+
+  for script in newScripts when !(nodeInList(script, oldScripts))
```

Comment:
I originally wrote it this way because I wasn't sure what browser support turbograft was aiming for or anything, but I think I should just yolo it because we're the only people who use it and nowhere on the readme does it mention any browsers :P